### DISCUSSION TITLE
Is fp16 logging broken?

### DISCUSSION THREAD

**Discussion by jpreiss** [bitcraze/discussions#1362] [Q&A]
Is `LOG_FP16` (half-precision float) logging broken? I am trying to use it and getting only zero bits back over the radio.

I couldn't quite figure out if `LOG_ADD(LOG_FP16, ...)` is meant to take the address of a `__fp16`, or if it takes a regular `float` and does the conversion internally, so I tried both - zero bits both times.

I noticed this switch statement doesn't have an entry for FP16:
https://github.com/bitcraze/crazyflie-firmware/blob/a25e8fb2254f6d5be54ad57566b3b846a97de93b/src/modules/src/log.c#L759

Also there is nowhere in `crazyflie-firmware/src` actually uses `LOG_FP16`, so I guess it has not been tested recently.

**Comment by knmcguire** [comment]
Hmm intersting and no we hardly test FP16 ourselves.

What we do test regurally is the basicLog, which fetches [one of the float ](https://github.com/bitcraze/crazyflie-firmware/blob/master/src/hal/src/pm_stm32f4.c#L510)as a FP16: https://github.com/bitcraze/crazyflie-lib-python/blob/c7eaabe31612e796fcdcf8511c18d197051e17cd/examples/logging/basiclog.py#L82. And that seems to always work.

Perhaps we should retire the LOG_FP16 entirely? We can convert this discussion to an issue for the crazyflie firmware

**Reply by jpreiss** [reply]
hmm, is it possible that `LOG_FP16` only supports this use case where the client says it wants to grab a `LOG_FLOAT` in reduced precision, but does *not* support declaring a log variable as `LOG_FP16` in `LOG_ADD()` directly?

**Reply by tobbeanton** [reply]
Yes, this might actually be the case. Does it work if you declare it as LOG_FLOAT?
